src/land/imsfv3_scf2ioda.py

```diff
@@ -0,0 +1,170 @@
+        lons = ncd.variables['lon'][:]
+        lats = ncd.variables['lat'][:]
+        oros = ncd.variables['oro'][:]
+        sncv = ncd.variables['IMSscf'][:]
+        sndv = ncd.variables['IMSsnd'][:]
+
+        lons = lons.astype('float32')
+        lats = lats.astype('float32')
+        oros = oros.astype('float32')
+        sncv = sncv.astype('float32')
+        sndv = sndv.astype('float32')
+
+        qcflg = 0*sncv.astype('int32')
+        qdflg = 0*sndv.astype('int32')
+        errsc = 0.0*sncv
+        errsd = 0.0*sndv
+        errsd[:] = 80.0
+        ncd.close()
+
+        times = np.empty_like(sncv, dtype=object)
+
+        # get datetime from filename
+        str_date = re.search(r'\d{8}', self.filename).group()
+        my_date = datetime.strptime(str_date, "%Y%m%d")
+        start_datetime = my_date.strftime('%Y-%m-%d')
+        base_datetime = start_datetime + 'T18:00:00Z'
+        AttrData['date_time_string'] = base_datetime
+
+        for i in range(len(lats)):
+            times[i] = base_datetime
+
+        # add metadata variables
+        self.outdata[('datetime', 'MetaData')] = times
+        self.outdata[('latitude', 'MetaData')] = lats
+        self.outdata[('longitude', 'MetaData')] = lons
+        self.outdata[('altitude', 'MetaData')] = oros
+
+        # add output variables
+        for i in range(len(sncv)):
+            for iodavar in ['snowCoverFraction', 'totalSnowDepth']:
+                if iodavar == 'snowCoverFraction':
+                    self.outdata[self.varDict[iodavar]['valKey']] = sncv
+                    self.outdata[self.varDict[iodavar]['errKey']] = errsc
+                    self.outdata[self.varDict[iodavar]['qcKey']] = qcflg
+                if iodavar == 'totalSnowDepth':
+                    self.outdata[self.varDict[iodavar]['valKey']] = sndv
+                    self.outdata[self.varDict[iodavar]['errKey']] = errsd
+                    self.outdata[self.varDict[iodavar]['qcKey']] = qdflg
+        DimDict['nlocs'] = len(self.outdata[('datetime', 'MetaData')])
+        AttrData['nlocs'] = np.int32(DimDict['nlocs'])
+
+
+def main():
+
+    parser = argparse.ArgumentParser(
+        description=('Read imsFV3 snow cover and depth file(s) and Converter'
+                     ' of native NetCDF format for observations of snow'
+                     ' cover and depth from imsFV3 to IODA netCDF format.')
+    )
+    parser.add_argument('-i', '--input',
+                        help="name of imsFV3 snow input file(s)",
+                        type=str, required=True)
+    parser.add_argument('-o', '--output',
+                        help="name of ioda output file",
+                        type=str, required=True)
+
+    args = parser.parse_args()
+
+    # Read in the imsFV3 snow data
+    ims = imsFV3(args.input)
+
+    # setup the IODA writer
+    writer = iconv.IodaWriter(args.output, locationKeyList, DimDict)
+
+    ims.varAttrs[('altitude', 'MetaData')]['units'] = 'm'
```

Comment:
To be consistent, I really suggest moving the assignment of units roughly to line#125 because this is odd to set one variables units here and all the others are done elsewhere.  Furthermore, can you rename ```altitude``` to ```height```?  This was a last-minute change to the variable naming conventions as discussed with UK Met Office.

Reply:
Yes, moved that sentence to there and renamed altitude to height as suggested.